tooling/templatize/pkg/pipeline/image_mirror.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/Azure/ARO-Tools/pkg/types"
+)
+
+func runImageMirrorStep(ctx context.Context, step *types.ImageMirrorStep, options *PipelineRunOptions, inputs map[string]Output, outputWriter io.Writer) error {
+	tmpFile, err := os.CreateTemp("", "")
+	if err != nil {
+		return fmt.Errorf("error creating script temp file %w", err)
+	}
+
+	err = os.Chmod(tmpFile.Name(), 0755)
```

Comment:
perfer to inline these in the `if` block to not leak `err` to outer scope (comment applies a couple times)

```go
if err := thing(); err != nil { }
```